packages/amplify-graphql-transformer-core/src/transformation/transform.ts

```diff
@@ -822,4 +825,12 @@ export class GraphQLTransform {
       index++;
     }
   }
+
+// Checks if DataStore is enabled and CPK Feature Flag is true. Throws a warning otherwise.
+public validateCPKFeatureFlag = (context: TransformerContext) => {
+  const isDataStoreEnabled = context.isProjectUsingDataStore();
+  if (isDataStoreEnabled && !isCPKFeatureEnabled(context)) {
+    console?.warn(`WARNING: Your schema has a custom primary key but the Feature Flag "${cpkFeatureFlagName}" is disabled. Check the value in your "amplify/cli.json" file, change it to "true" and re-run.`);
```

Comment:
DataStore being enabled without the CPK feature flag doesn't mean the schema is using a custom primary key. We should only throw the error if the schema actually contains a custom primary key.

Reply:
oh wow, good catch David! Will change accordingly 🤦🏼‍♀️

Reply:
updated with validating hasprimarykey() or not :)